review/get_review_ids.py

```diff
@@ -0,0 +1,212 @@
+'''
+Description
+'''
+
+import logging
+
+from courseware.models import StudentModule
+from opaque_keys.edx.locator import CourseLocator
+from opaque_keys.edx.keys import UsageKey
+from enrollment.api import get_enrollment, add_enrollment, update_enrollment
+from lms.djangoapps.course_blocks.api import get_course_blocks
+from xmodule.modulestore.django import modulestore
+import crum
+import random
+import json
+
+log = logging.getLogger(__name__)
+
+# TODO: Switch to using CourseLocators and/or CourseKeys everywhere
+REVIEW_COURSE_MAPPING = {
+    # Course used for testing. DO NOT REMOVE
+    'DillonX/DAD101x/3T2017': 'DillonX/DAD101rx/3T2017',
+    # Anant's course
+    'course-v1:MITx+6.002.3x+2T2016': 'MITx+6.002.3rx+2T2016',
+}
+ENROLLMENT_COURSE_MAPPING = {
+    # Course used for testing. DO NOT REMOVE
+    'DillonX/DAD101x/3T2017': 'DillonX/DAD101rx/3T2017',
+    # Anant's course
+    'course-v1:MITx+6.002.3x+2T2016': 'course-v1:MITx+6.002.3rx+2T2016',
+}
+TEMPLATE_URL = 'https://courses.edx.org/xblock/block-v1:{course_id}+type@{type}+block@{xblock_id}'
+
+
+def get_problems(num_desired, current_course):
+    '''
+    Looks through all the problems a learner has previously loaded and randomly
+    selects num_desired of them. Also checks if the learner had originally
+    answered it correctly or incorrectly and after how many attempts.
+
+    Parameters:
+        num_desired (int): the number of desired problems to show the learner
+        current_course (CourseLocator): The course the learner is currently in
+
+    Returns a list of num_desired tuples in the form (URL to display, correctness, attempts)
+    '''
+    user = crum.get_current_user()
+
+    enroll_user(user, current_course)
+    
+    problem_data = []
+    # Each record corresponds to a problem the user has loaded
+    # in the original course
+    for record in StudentModule.objects.filter(**{'student_id': user.id, 'course_id': current_course, 'module_type': 'problem'}):
+        # record.module_state_key takes the form: 
+        # block-v1:{course_id}+type@problem+block@{problem_id}
+        problem_id = str(record.module_state_key).split("@")[-1]
```

Comment:
The opaque-keys library can provide this problem ID info without having to serialize/parse the string, no?